src/main/java/com/mongodb/spark/sql/connector/read/MongoMicroBatchPartitionReader.java

```diff
@@ -0,0 +1,162 @@
+  private final ReadConfig readConfig;
+  private boolean closed = false;
+  private MongoClient mongoClient;
+  private MongoChangeStreamCursor<BsonDocument> changeStreamCursor;
+  private InternalRow currentRow;
+
+  /**
+   * Construct a new instance
+   *
+   * @param partition the partition
+   * @param bsonDocumentToRowConverter the converter from {@link BsonDocument} to {@link
+   *     InternalRow}
+   * @param readConfig the read configuration for reading from the partition
+   */
+  public MongoMicroBatchPartitionReader(
+      final MongoMicroBatchInputPartition partition,
+      final BsonDocumentToRowConverter bsonDocumentToRowConverter,
+      final ReadConfig readConfig) {
+
+    this.partition = partition;
+    this.bsonDocumentToRowConverter = bsonDocumentToRowConverter;
+    this.readConfig = readConfig;
+    LOGGER.info(
+        "Creating partition reader for: PartitionId: {} with Schema: {}",
+        partition.getPartitionId(),
+        bsonDocumentToRowConverter.getSchema());
+  }
+
+  /** Proceed to next record, returns false if there is no more records. */
+  @Override
+  public boolean next() {
+    Assertions.ensureState(() -> !closed, () -> "Cannot call next() on a closed PartitionReader.");
+    MongoChangeStreamCursor<BsonDocument> cursor = getCursor();
+    BsonDocument next = cursor.tryNext();
+    if (next != null) {
+      if (readConfig.streamPublishFullDocumentOnly()) {
+        next = next.getDocument(FULL_DOCUMENT, new BsonDocument());
+      }
+      currentRow = bsonDocumentToRowConverter.toInternalRow(next);
+    }
+    return next != null;
+  }
+
+  /** Return the current record. This method should return same value until `next` is called. */
+  @Override
+  public InternalRow get() {
+    Assertions.ensureState(() -> !closed, () -> "Cannot call get() on a closed PartitionReader.");
+    return currentRow;
+  }
+
+  @Override
+  public void close() {
+    if (!closed) {
+      closed = true;
+      releaseCursor();
+    }
+  }
+
+  private MongoChangeStreamCursor<BsonDocument> getCursor() {
+    if (mongoClient == null) {
+      mongoClient = readConfig.getMongoClient();
+    }
+    if (changeStreamCursor == null) {
+
+      List<BsonDocument> pipeline = new ArrayList<>();
+      pipeline.add(
+          Aggregates.match(Filters.lt("clusterTime", partition.getEndOffsetTimestamp()))
+              .toBsonDocument());
+
+      if (readConfig.streamPublishFullDocumentOnly()) {
+        pipeline.add(Aggregates.match(Filters.exists(FULL_DOCUMENT)).toBsonDocument());
+      }
+      pipeline.addAll(partition.getPipeline());
+
+      ChangeStreamIterable<Document> changeStreamIterable =
+          mongoClient
+              .getDatabase(readConfig.getDatabaseName())
+              .getCollection(readConfig.getCollectionName())
+              .watch(pipeline)
+              .fullDocument(readConfig.getStreamFullDocument())
+              .startAtOperationTime(partition.getStartAtOperationTime());
```

Comment:
As discussed, let's consider whether this should start at 0, or unspecified (the latter is closer to what we do for continuous stream).

Reply:
Added SPARK-378 to track this.

Now defaults to -1 and will only set startAtOperationTime if >= 0. So eventually users will be able to chose.